src/librustc_resolve/lib.rs

```diff
@@ -117,6 +119,27 @@ enum SuggestionType {
     NotFound,
 }
 
+/// Which candidates are shown: only the accessible ones,
+/// or also the not accessible ones. By default, we only
+/// show accessible candidates (`Accessible*` cases). If no
+/// accessible candidates exist, we show not accessible ones,
+/// `NotAccessible`. If, in the worst case, no candidates
+/// are found, we mark it as `NoCandidatesAtAll`
+pub enum CandidateAccessibility {
```

Comment:
Will do. I'm now addressing @nikomatsakis' feedback and will handle this immediately afterwards.